pkg/scheduler/objects/application.go

```diff
@@ -61,7 +62,8 @@ type Application struct {
 	allocatedResource *resources.Resource    // total allocated resources
 	allocations       map[string]*Allocation // list of all allocations
 	stateMachine      *fsm.FSM               // application state machine
-	stateTimer        *time.Timer            // timer for state time
+	startingTimer     *time.Timer            // timer for state time
+	waitingTimer      *time.Timer            // timer for state time
```

Comment:
Can we use 1 timer? I like the previous stateTimer better.
since this is a state machine, each time the app must be in one and only one state
we just need to make sure the timer is set properly when entering the state and reset when it exits from the state. we can set different duration for the different state.